### DISCUSSION TITLE
Is it normal to have the feeling that my code is too complicated?

### DISCUSSION THREAD

**Discussion by McMagnific** [codewars/codewars.com#2946] [General]
Hey Guys, I look forward to your replies!

So, I am coding for a little while now, and I would say, that I am maybe an advanced beginner.
And while I was working on a Keta to succeed it, I had the feeling that my approach and implementation was way too complicated
BUT at the end of the day the code works and the keta was done.

So far, so good.

But sometimes these thoughts of overcomplicating the approach makes me doubt that I am maybe not "thinking right" and it slows my workflow over all.

So is it a normal state of thinking about it and I just have to avoid the "imposter syndrome" which could come with it, or do you guys have a certain approach to get into a better workflow?

**Comment by JohanWiltink** [comment]
It is easier to remove complexity than to not write it in the first place. So I wouldn't say there's anything wrong with your programming. In time, you may be able to remove complexity earlier in the process, but there is nothing wrong with iterating on ( refactoring ) your solutions, occasionally even going back and changing the design.

Thinking about design before starting to crank out code is certainly useful, but you don't have to invent the perfectly round wheel before starting to code it. You know what they say about premature optimisation right?

**Reply by McMagnific** [reply]
Thanks for mention premature optimization. Never thought about that before.

So, you think that the "get it done first" approach is an approach I should internalize? 😅

**Reply by Evavic44** [reply]
I agree with Johan. It's the same reason Codewars allows you to improve your solution after completing it. Refactoring is a crucial part of programming.

**Reply by IlyaKrapivnoy** [reply]
And just after refactoring Codewars allows you to see refactored code of others and you realize that there is still a room for a bit more refactoring. I guess this is another side of the same question.

**Comment by raymondobroon** [comment]
I've only just started using CodeWars and the thing which makes it great for me is to see other developers solutions to the problems after my attempt has passed.

My initial thoughts on my version of the solution are usually that it is too basic / unpolished and seeing others more efficient code helps me to learn and improve.

I think the fact your solutions work shows that you have the skills in the first place but being able to compare and learn from others should help you to feel better about your efforts and ease any imposter syndrome jitters!

**Reply by jroycodes** [reply]
Totally agree with you. Seeing how others solved a problem only after you have solved it in your own way is a great deal for me. It gives room for improvement.

**Reply by ChairmanMawh** [reply]
> seeing others more efficient code

Just make sure to test this to actually be true. I see loads of solutions that are marked "best practices" or "clever" that I wouldn't describe as either. 

Also, typically solutions that truly are "Clever" would rarely be used in production because they are often unmaintainable or difficult to understand. Something that truly is a Best Practice would be a sensible approach to employ for real, but as noted, just because people vote a solution as BP doesn't mean it is
